### PR TITLE
feat: CLIN-1550 - export as TSV - with all

### PR DESCRIPTION
**Pull request by creativeyann17**
**New implementation of export as TSV feature**

This implementation allows to download even what's not displayed in the current page when user select **all** rows.

From a performance point of view, the requests to list the elements to download are always done but the limit size is 0 if nothing selected (good workaround to my opinion).

Screenshot from 2022-11-23 12-37-20

### CODE REVIEW

File: src/views/Prescriptions/Search/index.tsx [hunk 1]
```diff
@@ -46,6 +51,10 @@ const DEFAULT_SORT = [
     field: 'created_on',
     order: 'desc',
   },
+  {
```
Comment (creativeyann17) [inline src/views/Prescriptions/Search/index.tsx:54]: I noticed two refresh may not display the same order of rows if the same date, adding _id will fix that

File: src/views/Prescriptions/Search/components/table/PrescriptionTable/index.tsx [hunk 2]
```diff
@@ -77,12 +66,19 @@ const PrescriptionsTable = ({
           total: results?.total || 0,
         },
         enableColumnSort: true,
-        onSelectedRowsChange: (e) => {
-          setSelectedKeys(e);
+        onSelectedRowsChange: (keys) => {
+          setSelectedKeys(keys);
+        },
```
Comment (francisl) [inline src/views/Prescriptions/Search/components/table/PrescriptionTable/index.tsx:71]:
```
onSelectedRowsChange: setSelectedKeys,
```
tu peux lui passer la fonction directement dans ce cas, pas besoin de wrapper

File: src/views/Prescriptions/Search/components/table/SequencingTable/index.tsx [hunk 3]
```diff
@@ -77,12 +66,19 @@ const SequencingsTable = ({
...
+          setSelectedKeys([ALL_KEYS]);
         },
         enableTableExport: true,
         onTableExportClick: () => {
-          download(results, selectedKeys);
+          if (selectedKeys.length === 0) {
+            setDownloadKeys([ALL_KEYS]);
+          } else {
+            setDownloadKeys(selectedKeys);
+          }
```
Comment (francisl) [inline src/views/Prescriptions/Search/components/table/SequencingTable/index.tsx:81]: else superflu

File: src/views/Prescriptions/Search/components/table/SequencingTable/index.tsx [hunk 4]
```diff
@@ -77,12 +66,19 @@ const SequencingsTable = ({
           total: results?.total || 0,
         },
         enableColumnSort: true,
-        onSelectedRowsChange: (e) => {
-          setSelectedKeys(e);
+        onSelectedRowsChange: (keys) => {
+          setSelectedKeys(keys);
+        },
```
Comment (francisl) [inline src/views/Prescriptions/Search/components/table/SequencingTable/index.tsx:71]: ☝️

File: src/views/Prescriptions/Search/index.tsx [hunk 5]
```diff
@@ -142,6 +220,9 @@ const PrescriptionSearch = (): React.ReactElement => {
                 results={prescriptions}
                 queryConfig={prescriptionQueryConfig}
                 setQueryConfig={setPrescriptionQueryConfig}
+                setDownloadKeys={(keys) => {
+                  setDownloadPrescriptionKeys(keys);
+                }}
```
Comment (francisl) [inline src/views/Prescriptions/Search/index.tsx:225]: ☝️

File: src/views/Prescriptions/Search/index.tsx [hunk 6]
```diff
@@ -159,6 +240,9 @@ const PrescriptionSearch = (): React.ReactElement => {
                 results={sequencings}
                 queryConfig={sequencingQueryConfig}
                 setQueryConfig={setSequencingQueryConfig}
+                setDownloadKeys={(keys) => {
+                  setDownloadSequencingKeys(keys);
+                }}
```
Comment (francisl) [inline src/views/Prescriptions/Search/index.tsx:245]: ☝️

File: src/views/Prescriptions/utils/export.ts [hunk 7]
```diff
@@ -1,28 +1,52 @@
...
-    ? data.filter((row) => selectedKeys.includes(row[key]))
-    : data;
+): any[] => {
+  if (dataKeys.length === 0) {
+    return [];
+  } else if (dataKeys.length === 1 && dataKeys[0] === ALL_KEYS) {
+    return data;
+  } else {
+    return data.filter((row) => dataKeys.includes(row[key]));
+  }
```
Comment (francisl) [inline src/views/Prescriptions/utils/export.ts:48]: else superflu

**Review by francisl — APPROVED** [review]
non blocking comments.

**Pull request merged**